File: Algebra/gcd/python/gcd.py
"""
    euclid_gcd(0 is used to find the gcd of two numbers using euclidean algorithm
    @input: num1, num2 numbers
    @output gcd(num1,num2)
"""

def euclid_gcd(num1, num2):
    
    if num1 == 0:
        return num2    
    if num2 == 0 :
        return num1    
    if num1 < num2:
        return euclid_gcd(num2, num1)
    while num2 != 0:
        (num1, num2) = (num2, num1%num2)
    return num1

File: Algebra/gcd/cpp/gcd.cpp
#include <iostream>
using namespace std;

int main(){
    int num1, num2;
    cout << "Enter two numbers: ";
    cin >> num1 >> num2;   
    while(num1 != num2){
        if(num1 > num2)
            num1 -= num2;
        else
            num2 -= num1;
    }
    cout << "HCF = " << num1;
    return 0;
}
